internal/pkg/cli/pipeline_list.go

```diff
@@ -75,60 +107,157 @@ func (o *listPipelineOpts) Ask() error {
+	for _, pipeline := range deployed {
+		cp[pipeline.Name] = pipeline
+	}
+
+	type combinedInfo struct {
+		Name         string `json:"name"`
+		ManifestPath string `json:"manifestPath"`
+		*codepipeline.Pipeline
+	}
+
+	var out struct {
+		Pipelines []combinedInfo `json:"pipelines"`
+	}
+	for _, pipeline := range local {
+		out.Pipelines = append(out.Pipelines, combinedInfo{
+			Name:         pipeline.Name,
+			ManifestPath: pipeline.Path,
+			Pipeline:     cp[pipeline.Name],
+		})
+	}
+
+	b, err := json.Marshal(out)
+	if err != nil {
+		return fmt.Errorf("marshal pipelines: %w", err)
+	}
+
+	fmt.Fprintf(o.w, "%s\n", b)
+	return nil
+}
+
+// humanOutputLocal prints the name of all pipelines in the current workspace.
+func (o *listPipelineOpts) humanOutputLocal() error {
+	local, err := o.workspace.ListPipelines()
+	if err != nil {
+		return err
+	}
+
+	for _, pipeline := range local {
+		fmt.Fprintln(o.w, pipeline.Name)
 	}
-	fmt.Fprint(o.w, out)
 
 	return nil
 }
 
-func (o *listPipelineOpts) jsonOutput(pipelines []*codepipeline.Pipeline) (string, error) {
+// jsonOutputDeployed prints data about all pipelines in the given app that have been deployed.
+func (o *listPipelineOpts) jsonOutputDeployed(ctx context.Context) error {
+	pipelines, err := getDeployedPipelines(ctx, o.appName, o.pipelineLister, o.codepipeline)
+	if err != nil {
+		return err
+	}
+
 	type serializedPipelines struct {
 		Pipelines []*codepipeline.Pipeline `json:"pipelines"`
 	}
 	b, err := json.Marshal(serializedPipelines{Pipelines: pipelines})
 	if err != nil {
-		return "", fmt.Errorf("marshal pipelines: %w", err)
+		return fmt.Errorf("marshal pipelines: %w", err)
 	}
-	return fmt.Sprintf("%s\n", b), nil
+
+	fmt.Fprintf(o.w, "%s\n", b)
+	return nil
 }
 
-func (o *listPipelineOpts) humanOutput(pipelines []string) string {
-	b := &strings.Builder{}
-	for _, pipeline := range pipelines {
-		fmt.Fprintln(b, pipeline)
+// humanOutputDeployed prints the name of all pipelines in the given app that have been deployed.
+func (o *listPipelineOpts) humanOutputDeployed() error {
+	pipelines, err := o.pipelineLister.ListDeployedPipelines(o.appName)
+	if err != nil {
+		return fmt.Errorf("list deployed pipelines: %w", err)
 	}
-	return b.String()
+
+	sort.Slice(pipelines, func(i, j int) bool {
+		return pipelines[i].Name < pipelines[j].Name
+	})
+
+	for _, p := range pipelines {
+		fmt.Fprintln(o.w, p.Name)
+	}
+
+	return nil
+}
+
+func getDeployedPipelines(ctx context.Context, app string, lister deployedPipelineLister, getter pipelineGetter) ([]*codepipeline.Pipeline, error) {
+	pipelines, err := lister.ListDeployedPipelines(app)
+	if err != nil {
+		return nil, fmt.Errorf("list deployed pipelines: %w", err)
+	}
+
+	var mux sync.Mutex
+	var res []*codepipeline.Pipeline
+
+	g, _ := errgroup.WithContext(ctx)
+
+	for i := range pipelines {
+		pipeline := pipelines[i]
+		g.Go(func() error {
+			info, err := getter.GetPipeline(pipeline.ResourceName)
```

Comment:
Do you know how long it usually takes to make one API call like this 🤔

Reply:
Just ran a few tests - usually around 150-250 ms per call.